hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,514 @@
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        accepted = True
+        publish_template = self._get_template("Publish file Template", settings)
+        if not publish_template and not item.parent.properties["work_file_template"]:
+            accepted = False
+
+        # check that the AbcExport command is available!
+        if not mel.eval("exists \"AbcExport\""):
+            accepted = False
+            self.logger.warn("Item not accepted because export command AbcExport is not available.")
+        
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        # check that there is still geometry in the scene:
+        if not cmds.ls(geometry=True, noIntermediate=True):
+            self.logger.warn("Validation failed because there is no geometry in the scene to be exported")
+            return False
+
+        publisher = self.parent
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # get the configured work file template
+        work_template = item.parent.properties["work_file_template"]
+        publish_template = self._get_template("Publish file Template", settings)
+
+        # get the current scene path and extract fields from it
+        # using the work template:
+        scene_path = os.path.abspath(cmds.file(query=True, sn=True))
```

Comment:
In the other hooks we use `sgtk.util.ShotgunPath.normalize(path)` to ensure its formatted in a way that shotgun expects. You can see it here: https://github.com/shotgunsoftware/tk-maya/pull/57/files#diff-ec4ba2e3fd4512d2c469433269615506R218. There's also a reused `_session_path` method that does some additonal sanitization. I know it stinks to duplicate it, but I suggest adding it here as well.

Reply:
Ok. Added it